### ISSUE TITLE
Call activity does not honor the lane assignment

### ISSUE THREAD

**Issue by stefanroels** [sartography/spiff-arena#1131]
I have made a process which (depending on a condition) needs to assign a task to one of two groups. The task itself is the same, so I have made a separate bpmn (`subtask`) file containing the actions (which includes a user action) and in the main bpmn (`maintask`) I have created two lanes that each have a call activity referring to `subtask`.

My expectation was that, as the call activity is in a lane, the corresponding group would be required to perform the action, but it appears that, as the tasks in `subtask` are not bound to any lane itself, the original user (which is not part of the lane group) is presented with the action.

My expectations would be:
* If a (user or manual) task is outside a lane, the current user gets the task
* If a (user or manual) task is inside a lane, users in the lane group get the task
* If a call activity is in a lane, users in the lane group get the tasks in the called activity, if the call activity does not specify its own lanes
* If a call activity is in a lane, and the called activity specifies a lane, that lane overrides the group assignment of the call activity.

By the way, the main reason I got into this mess is that I wanted to have an approval process (any user can start a request, a certain group of people can grant approval). The difficulty was that if the person that starts the request is in the group that can grant approval, the user shall not be allowed to approve their own request. Unfortunately, there is no way (as far as I have seen) to assign tasks to specific people (something like: assigned = get_group_members('approvers'); assigned.remove(get_process_initiator_user() )

**Comment by burnettk** [comment]
regarding the get group members concept at the end, from a recent discord discussion:

let's say you have a lane called "warehouse". let's also say that in a given process instance, there is data set such that participant_id is "participant_b" and branch_id is "branch_a". in that case, in a script task before the relevant user task, you could write some code like the following. this would result in all of the members of the group "branch_a_participant_b" being assigned to tasks in the "warehouse" lane: 

```python
group_identifier = f"{branch_id}_{participant_id}"
group_members = get_group_members(group_identifier)
lane_owners = { "warehouse": group_members }
```

so i think you actually could call `get_process_initiator_user()` , get the username, and subtract that user from the list of users to assign to the lane owners dictionary for the lane in question. hopefully this is somewhat clear.

**Comment by stefanroels** [comment]
I did find this trick last week. As far as I have seen, the `lane_owners` is not documented anywhere, so I was not aware of this until I stumbled across it.

**Comment by burnettk** [comment]
FYI, @usama9500 , a documentation opportunity here with lane_owners. Glad you found it,  presumably in the source.

I will assume you are unblocked but will leave this open until we can clarify your four assumptions and which of them are working as we collectively expect, and whether we need any code changes to make things work more coherently.

**Comment by burnettk** [comment]
I think it is just documentation, and it is complete. Regarding the four assumptions above, I haven’t verified them thoroughly, but I believe they are all correct except for number three. That is, the Parent call activity is ignored, and the user task itself must be in a lane in the called activity. Closing, But feel free to reopen.

**Comment by calexh-sar** [comment]
Reopening and setting Status to Documenation.  @usama9500 please close one you have finished your work on this issue.

**Comment by usama9500** [comment]
@calexh-sar I have documented generically assigning lane owners section, how it wokrs and how lanes are assigned [here](https://spiff-arena.readthedocs.io/en/latest/Building_Diagrams/pools_and_lanes.html#assigning-lane-owners). For the [last comment](https://github.com/sartography/spiff-arena/issues/1131#issuecomment-2027559969), I will add the "get group members" details and how to perform the function.

**Comment by usama9500** [comment]
Documented get group members here: https://github.com/sartography/spiff-arena/pull/1631
